File: Periodic Practice1/Multiplication.py
n = int(input())

a = list(map(int, input().split()))
b = list(map(int, input().split()))

a = [(a[i], i) for i in range(n)]

a.sort()
b.sort()

ans = [0] * n
total = 0

for i in range(n):
    value, index = a[i]

    ans[index] = b[i]
    total += value * b[i]

print(total)
print(*ans)

File: Periodic Practice1/Multiplication.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<pair<int, int>> a(n);
    vector<int> b(n);

    for (int i = 0; i < n; i++) {
        cin >> a[i].first;
        a[i].second = i;
    }

    for (int i = 0; i < n; i++) {
        cin >> b[i];
    }

    sort(a.begin(), a.end());

    sort(b.begin(), b.end());

    long long total = 0;
    vector<int> ans(n);

    for (int i = 0; i < n; i++) {
        int index = a[i].second;

        ans[index] = b[i];

        total += 1LL * a[i].first * b[i];
    }

    cout << total << '\n';

    for (int i = 0; i < n; i++) {
        cout << ans[i] << ' ';
    }
}